server/mscalendar/daily_summary.go

```diff
@@ -0,0 +1,189 @@
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils"
+	"github.com/mattermost/mattermost-plugin-mscalendar/server/utils/tz"
+	"github.com/pkg/errors"
+)
+
+const dailySummaryTimeWindow = time.Minute * 2
+
+var timeNowFunc = time.Now
+
+type DailySummary interface {
+	GetDailySummarySettingsForUser(user *User) (*store.DailySummarySettings, error)
+	SetDailySummaryPostTime(user *User, timeStr string) (*store.DailySummarySettings, error)
+	SetDailySummaryEnabled(user *User, enable bool) (*store.DailySummarySettings, error)
+	DailySummaryAll() error
+}
+
+func (m *mscalendar) GetDailySummarySettingsForUser(user *User) (*store.DailySummarySettings, error) {
+	dsumIndex, err := m.Store.LoadDailySummaryIndex()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dsum := range dsumIndex {
+		if dsum.MattermostUserID == user.MattermostUserID {
+			return dsum, nil
+		}
+	}
+
+	return nil, errors.New("No daily summary settings found")
+}
+
+func (m *mscalendar) SetDailySummaryPostTime(user *User, timeStr string) (*store.DailySummarySettings, error) {
+	t, err := time.Parse(time.Kitchen, timeStr)
+	if err != nil {
+		return nil, errors.New("Invalid time value")
+	}
+
+	if t.Minute() != 0 && t.Minute() != 30 {
+		return nil, errors.New("Time must be a multiple of 30 minutes.")
+	}
+
+	timezone, err := m.GetTimezone(user)
+	if err != nil {
+		return nil, err
+	}
+
+	dsumIndex, err := m.Store.LoadDailySummaryIndex()
+	if err != nil {
+		return nil, err
+	}
+
+	var result *store.DailySummarySettings
+	for _, dsum := range dsumIndex {
+		if dsum.MattermostUserID == user.MattermostUserID {
+			dsum.PostTime = timeStr
+			dsum.Timezone = timezone
+			result = dsum
+			break
+		}
+	}
+	if result == nil {
+		result = &store.DailySummarySettings{
+			MattermostUserID: user.MattermostUserID,
+			PostTime:         timeStr,
+			Timezone:         timezone,
+		}
+		dsumIndex = append(dsumIndex, result)
+	}
+
+	err = m.Store.SaveDailySummaryIndex(dsumIndex)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (m *mscalendar) SetDailySummaryEnabled(user *User, enable bool) (*store.DailySummarySettings, error) {
+	err := m.Filter(
+		withClient,
+		withUserExpanded(user),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	dsumIndex, err := m.Store.LoadDailySummaryIndex()
+	if err != nil {
+		return nil, err
+	}
+
+	var result *store.DailySummarySettings
+	for _, dsum := range dsumIndex {
+		if dsum.MattermostUserID == user.MattermostUserID {
+			dsum.Enable = enable
+			result = dsum
+		}
+	}
+	if result == nil {
+		result = &store.DailySummarySettings{
+			MattermostUserID: user.MattermostUserID,
+			Enable:           enable,
+		}
+		dsumIndex = append(dsumIndex, result)
+	}
+
+	err = m.Store.SaveDailySummaryIndex(dsumIndex)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func (m *mscalendar) DailySummaryAll() error {
```

Comment:
Naming here feels weird. I would go for a "PostAllDailySummary" or just "PostDailySummary".

Reply:
I'm weary of using `Post` in the name, because most of the time the job runs, there will be no posts. It makes me think we are force-posting to all users, regardless if it is at their saved time. @levb Do you have any thoughts on this?

Reply:
Not sure of the full context, but maybe `Generate...`?